### PR TITLE
[KeyVault] adding readme.md for data-plane

### PR DESCRIPTION
**Pull request by schaabs**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/KeyVault/dataPlane/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+The Microsoft Azure Key Vault SDK for .NET allows you to build secure Azure applications that can access secrets, keys, and certificates that a encrypted at rest with hardware security modules.
+
+This directory contains the open source subset of the .NET SDK. For documentation of the 
+complete Azure SDK, please see the [Microsoft Azure .NET Developer Center](http://azure.microsoft.com/en-us/develop/net/).
+
+## Features
+
+- Secrets
+    - Create, Read, Update, and Delete Secrets
```
Comment (dragav) [inline src/SDKs/KeyVault/dataPlane/README.md]: Do you want to mention recover as well? Same for keys and certs.

File: src/SDKs/KeyVault/dataPlane/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+The Key Vault Client Library shares the same [Prerequisites](https://github.com/azure/azure-sdk-for-net#prerequisites) as the Microsoft Azure SDK for .NET.
+
+## To Build
+
+For information on building the Azure Key Vault SDK, please see [Building the Microsoft Azure SDK for .NET](https://github.com/azure/azure-sdk-for-net#to-build)
+
+## Running Tests
+
+Tests for the Azure Key Vault SDK are run in the same manner as the rest of the tests for the Azure SDK for .NET.  For information please see, [to run the tests](https://github.com/azure/azure-sdk-for-net#to-run-the-tests)
```
Comment (dragav) [inline src/SDKs/KeyVault/dataPlane/README.md]: (nitpick) I would rephrase this to "please see how to run tests" (with the 'run tests' being hyperlinked).

File: src/SDKs/KeyVault/dataPlane/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+## Samples
+
+Code samples are available in the Azure Key Vault SDK code in the [Microsoft.Azure.KeyVault.Samples](Microsoft.Azure.KeyVault.Samples) directory.  Addictional samples can also be found on [Azure Code Samples](https://azure.microsoft.com/en-us/resources/samples/?sort=0&service=key-vault&platform=dotnet).
+
+## Additional Documentation
+
+* Azure Key Vault General Documentation - https://docs.microsoft.com/en-us/azure/key-vault/
```
Comment (dragav) [inline src/SDKs/KeyVault/dataPlane/README.md]: Would suggest making these hyperlinks

**Review by dragav — APPROVED** [review]

**Review by dragav — APPROVED** [review]

**Review by shahabhijeet — APPROVED** [review]
